### DISCUSSION TITLE
Audio Tracks management

### DISCUSSION THREAD

**Discussion by jocphx** [grizzlypeak3d/DJV#452] [Q&A]
Hi,

I am checking out DJV player, and I cannot find any menu to manage the audio tracks of my video clip. The Audio tab on the UI shows only volume increase, decrease and mute.

I have 24 audio tracks. Does DJV support more than 2 audio tracks?

**Comment by visualfeast** [comment]
Testing it out myself and everything is amazing, except the audio controls. I need to be able to pick which track(s) to monitor. Other than this, DJV is almost perfect.

**Reply by darbyjohnston** [reply]
Hi, and sorry for the long delay; I just released a new version, 3.0.0, that has support for multiple audio tracks. If you have a chance try it out and see what you think.

**Reply by visualfeast** [reply]
Hi, I can't test it here on Windows as this new 3.0.0 version doesn't open any of my videos. DJV at the bottom says "ffmpeg::Read: ERROR: No video codec found:" (then lists the filename).

Attached is the log from trying to open 2 different types that worked on 2.0.8.

Also, I see this line in the log, perhaps part of the issue?
`ffmpeg::ReadPlugin:  is not implemented. Update your FFmpeg version to the newest one from Git. If the problem still occurs, it means that your file has a feature which has not been implemented.`

[log 7-23-25.txt](https://github.com/user-attachments/files/21390940/log.7-23-25.txt)

Thanks

**Reply by darbyjohnston** [reply]
@visualfeast Sorry about the codecs, but I cannot include them in the binaries. If you don't mind building from source the codecs will be enabled by default.

**Reply by visualfeast** [reply]
Thanks @darbyjohnston. I tried following the build instructions, but the cmake fails. I tried v4.1.0 and v3.31.8. I haven't done this before and I don't know what I did wrong (output below). Any idea what I'm doing wrong and how to build this on Windows 10?

```
F:\DJV3>cmake -S DJV\etc\SuperBuild -B Release -DCMAKE_INSTALL_PREFIX=%CD%\Release\install -DCMAKE_PREFIX_PATH=%CD%\Release\install -DCMAKE_BUILD_TYPE=Release
-- Building for: NMake Makefiles
CMake Error at CMakeLists.txt:5 (project):
  Running

   'nmake' '-?'

  failed with:

   no such file or directory


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!

```

**Reply by darbyjohnston** [reply]
Hi, it looks like you might be using the regular command prompt instead of the Visual Studio command prompt. Open the start menu, find the Visual Studio folder, and open the command prompt called "x64 Native Tools Command Prompt for VS 2022". That will setup the environment with the build tools.

**Reply by visualfeast** [reply]
Yes, thanks! :) I didn't know I needed it, so installing Visual Studio now... Does it need any specific Workloads or Individual Components?

**Reply by darbyjohnston** [reply]
I think just C++ desktop development.

**Reply by visualfeast** [reply]
Thanks, with your help I got it compiled! Some things I noticed:

1) My MXF files (recorded on a Sony cinema camera which has up to 4 mono tracks of audio) are not giving me the option to mute certain tracks. It just has the channel mute for 0, which mutes all. Same issue with some ProRes. What's interesting is the ones that show 1 channel in DJV, show all the channels in MPC-HC, and those showing 4 in DJV only show 1 in MPC-HC.

2) Trying to close a video's tab before it finishes loading/caching, will lock up the player hard (sometimes it'll come back after a long time, but I usually give up & force close player after 30 sec or so). With some of the 50+ GB cam files, the thumbs/waveform loading can take quite a long time.

3) 1440x1080 (Panasonic DVCPro P2) files have a PAR of 0.75, and the player doesn't respect this or have any way of setting the aspect ratio, so all that video shows as squished.

4) I'd love an option to _not_ pre-load up waveform & thumbnails at all (so loading up a vid would be as fast as v2 or MPC), plus the waveforms are very slow on big videos, esp ones not on super-fast PCIE media. I tried adjusting the video & audio cache settings. When clicking over to another video's tab, it has to re-load the thumbs/waveforms again, which is sometimes faster than the first time, but usually same speed as original load.

More testing later-- thanks so much. Do I need to search existing issues for these or log new issues if they don't exist?

**Reply by darbyjohnston** [reply]
Nice, that's great you got it to build.

1. I'm not very familiar with MXF, it can contain multiple audio tracks?
2. I tried with a small test MXF and it closed OK. If there are a lot of frames stored in the memory cache (like GBs), it can take some time to free them when closing.
3. Do you have an example file? It's probably not picking up the right metadata for the aspect ratio.
4. You can turn off the thumbnails in the "Timeline" menu, that will help the performance a bit.

**Reply by visualfeast** [reply]
MXF can have tons of audio tracks, yes. Most of mine are around 20-80GB each, but I have some smaller 1-2GB ones-- can I send a sample to your email, shared through Google Drive?
